projects/adc_challenge/src/main.c

```diff
@@ -0,0 +1,63 @@
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+// reads from the specified address and logs the value
+static void prv_read_from_adc_addr(const GpioAddress *address, void *context) {
+  GpioAddress addr = {
```

Comment:
The GPIO address you passed in when you initialized the interrupt actually is passed in `void *context`
You will need to cast this to a `GpioAddress`

`const GpioAddress *address` is actually the address of the pin that triggers the interrupt